js_modules/dagit/packages/core/src/ui/SectionedLeftNav.tsx

```diff
@@ -0,0 +1,267 @@
+      }
+    }
+    return [...reposWithJobs, ...reposWithoutJobs];
+  }, [visibleRepos]);
+
+  if (loading) {
+    return <div style={{flex: 1}} />;
+  }
+
+  return (
+    <>
+      <Box
+        flex={{direction: 'row', alignItems: 'center', gap: 8}}
+        padding={{horizontal: 24, bottom: 12}}
+        border={{side: 'bottom', width: 1, color: Colors.KeylineGray}}
+      >
+        <Icon name="job" />
+        <span style={{fontSize: '16px', fontWeight: 600}}>Jobs</span>
+      </Box>
+      <Container>
+        {sortedRepos.map((repo) => {
+          const repoName = repo.repository.name;
+          const repoAddress = buildRepoAddress(repoName, repo.repositoryLocation.name);
+          const addressAsString = repoAddressAsString(repoAddress);
+          return (
+            <Section
+              key={addressAsString}
+              onToggle={onToggle}
+              option={repo}
+              expanded={expandedKeys.includes(addressAsString)}
+              showRepoLocation={duplicateRepoNames.has(repoName)}
+            />
+          );
+        })}
+      </Container>
+    </>
+  );
+};
+
+const HEADER_HEIGHT = 48;
+const HEADER_HEIGHT_WITH_LOCATION = 64;
+
+interface SectionProps {
+  expanded: boolean;
+  onToggle: (repoAddress: RepoAddress) => void;
+  option: DagsterRepoOption;
+  showRepoLocation: boolean;
+}
+
+export const Section: React.FC<SectionProps> = (props) => {
+  const {expanded, onToggle, option, showRepoLocation} = props;
+  const repoAddress = buildRepoAddress(option.repository.name, option.repositoryLocation.name);
+  const jobItems = React.useMemo(() => getJobItemsForOption(option), [option]);
+  const anyJobs = jobItems.length > 0;
+  const showJobs = expanded && anyJobs;
+
+  return (
+    <Box background={Colors.Gray100}>
+      <SectionHeader
+        $open={showJobs}
+        $showRepoLocation={showRepoLocation}
+        disabled={!anyJobs}
+        onClick={() => onToggle(repoAddress)}
+      >
+        <Box flex={{direction: 'row', alignItems: 'flex-start', gap: 8}}>
+          <Box margin={{top: 2}}>
+            <Icon name="folder_open" size={16} />
+          </Box>
+          <RepoNameContainer>
+            <div style={{minWidth: 0}}>
+              <RepoName style={{fontWeight: 500}} data-tooltip={option.repository.name}>
+                {option.repository.name}
+              </RepoName>
+              {showRepoLocation ? (
+                <RepoLocation
+                  data-tooltip={`@${option.repositoryLocation.name}`}
+                  $disabled={!anyJobs}
+                >
+                  @{option.repositoryLocation.name}
+                </RepoLocation>
+              ) : null}
+            </div>
+            {/* Wrapper div to prevent tag from stretching vertically */}
+            <div>
+              <BaseTag
+                fillColor={Colors.Gray10}
+                textColor={Colors.Dark}
+                label={new Intl.NumberFormat().format(jobItems.length)}
```

Comment:
I think we do this elsewhere with `Number(1000).toLocaleString()` , maybe they're short-hands for each other? (Actually need to read up on this version...)